db-tests/src/test/java/gov/nasa/jpl/aerie/database/PlanCollaborationTests.java

```diff
@@ -723,5 +968,1640 @@ void grandparentAdoptsChildrenOnDelete() throws SQLException{
+      for(int i = 75;  i < 100; ++i) { deleteActivityDirective(childPlan, baseActivities[i]); }
+      for(int i = 100; i < 150; ++i) { updateActivityName("Renamed Activity " + i, activityId, basePlan); }
+      for(int i = 150; i < 200; ++i) { updateActivityName("Renamed Activity " + i, activityId, childPlan); }
+      for(int i = 0;   i < 25;  ++i) { insertActivity(basePlan); }
+
+      final int mergeRQ = createMergeRequest(basePlan, childPlan);
+      beginMerge(mergeRQ);
+      commitMerge(mergeRQ);
+      //assert that all activities are included now
+      assertEquals(375, getActivities(basePlan).size()); //425-50
+    }
+
+    @Test
+    void modifyAndDeletesApplyCorrectly() throws SQLException {
+      /*
+      Checks:
+      -- modify uncontested supplying applies
+      -- delete uncontested supplying applies
+      -- modify contested supplying applies
+      -- modify contested receiving does nothing
+      -- delete contested supplying applies
+      -- delete contested receiving applies
+       */
+
+      final int basePlan = insertPlan(missionModelId);
+      final int modifyUncontestedActId = insertActivity(basePlan);
+      final int deleteUncontestedActId = insertActivity(basePlan);
+      final int modifyContestedSupplyingActId = insertActivity(basePlan);
+      final int modifyContestedReceivingActId = insertActivity(basePlan);
+      final int deleteContestedSupplyingActId = insertActivity(basePlan);
+      final int deleteContestedReceivingActId = insertActivity(basePlan);
+      final int childPlan = duplicatePlan(basePlan, "Child");
+
+      assertEquals(6, getActivities(basePlan).size());
+      assertEquals(6, getActivities(childPlan).size());
+
+      updateActivityName("Test", modifyUncontestedActId, childPlan);
+
+      deleteActivityDirective(childPlan, deleteUncontestedActId);
+
+      updateActivityName("Modify Contested Supplying Parent", modifyContestedSupplyingActId, basePlan);
+      updateActivityName("Modify Contested Supplying Child", modifyContestedSupplyingActId, childPlan);
+
+      updateActivityName("Modify Contested Receiving Parent", modifyContestedReceivingActId, basePlan);
+      updateActivityName("Modify Contested Receiving Child", modifyContestedReceivingActId, childPlan);
+
+      updateActivityName("Delete Contested Supplying Parent", deleteContestedSupplyingActId, basePlan);
+      deleteActivityDirective(childPlan, deleteContestedSupplyingActId);
+
+      deleteActivityDirective(basePlan, deleteContestedReceivingActId);
+      updateActivityName("Delete Contested Receiving Child", deleteContestedReceivingActId, childPlan);
+
+      final int mergeRQ = createMergeRequest(basePlan, childPlan);
+      beginMerge(mergeRQ);
+
+      assertEquals(4, getConflictingActivities(mergeRQ).size());
+      setResolution(mergeRQ, modifyContestedSupplyingActId, "supplying");
+      setResolution(mergeRQ, modifyContestedReceivingActId, "receiving");
```

Comment:
Did we standardize on `source` and `target`?

Reply:
My understanding of the situation was that the backend uses "supplying" and "receiving" while the front end displays "source" and "target". Or did we decide to standardize between the two?

Reply:
I see - I guess it would be good for the two to be consistent, but maybe not completely required.

The `setResolution` jumped out at me particularly because I was under the impression that it is visible via the api - when the client updates the conflicting_activities table to set a resolution - am I overlooking a translation step here?

Reply:
IIRC, the inconsistency arose because the DB wants to be clear while the UI abstracted for the sake of conciseness/not being ugly. Unless you mean that our Hasura functions should be consistent with the UI terminology? In which case, I don't disagree, although it does beg the question as to if we should add Hasura functions specifically to do this `source`->`supplying` `target`->`receiving` mapping, since currently the [example query for resolving conflicts](https://github.com/NASA-AMMOS/aerie/pull/399/commits/024b877a8361d63ea1ede21b5495113323f21c9b#diff-a0597b372569429c046ef7d6732fc976ac99296101ad8bdb066697775709d65cR409-R436) is just doing a straight table update (and therefore using the DB terms).

As for if you're missing a translation point, maybe? This `setResolution` is taking place in our `dbTests`. [The method](https://github.com/NASA-AMMOS/aerie/pull/399/files/024b877a8361d63ea1ede21b5495113323f21c9b?diff=unified&w=0#diff-269125a2369e062ea57b1a0ae448f5e159d3c8f69552c13ace30c65533e246b2R416-R426) queries the database directly rather than going through the API, which is why it's using the DB term.

Reply:
> the [example query for resolving conflicts](https://github.com/NASA-AMMOS/aerie/pull/399/commits/3795c44f8002c69f589284af866349aab517321c#diff-a0597b372569429c046ef7d6732fc976ac99296101ad8bdb066697775709d65cR409-R441) is just doing a straight table update (and therefore using the DB terms)

👆 yes - this is what I was thinking about - the "straight table update" is part of our api (since we're exposing the table via hasura), so I'm proposing that all client-visible terminology be consistent. 

>  it does beg the question as to if we should add Hasura functions specifically to do this

Yes - this is what I'm imagining a "translation point" would look like - though I would advocate for changing the DB terms to be consistent with the API terms rather than adding a function to translate between them. Is there an advantage to using one set for the API and a different set internally?

Reply:
Mainly clarity. One thing that has come up often during this work is that `source` and `target` _are_ confusing, while `snapshot_supplying_changes` and `plan_receiving_changes` keep it obvious what exactly something _is_ and what exactly it's _doing_ in the merge. Personally, I'd rather have the code be understandable at the cost of a translation point at either the UI or API level rather than consistent at all points, especially with something like merging, where the code is already complicated enough without adding in a prerequisite of having to keep clear what `source` and `target` are.

While we could change the definition of `conflict_resolution` to `{ none, source, target }`, this would just move the translation point to inside the database, which I'd argue is worse than having it on the communications interface.

A question I have: does the API _need_ to use the UI display terms?

Reply:
> A question I have: does the API need to use the UI display terms?

It technically doesn't - certainly if we ever add internationalization, the words displayed in the UI will differ from the terms in the API. However, there is a lot of benefit to having a [ubiquitous language](https://www.agilealliance.org/glossary/ubiquitous-language) that is shared among developers, product people, and customers - a shared vocabulary reduces friction and helps everyone be on the same page.

If we do go the route of adding a translation layer, we'll need to hide the `merge_staging_area` and `conflicting_activities` tables, and provide views that translate between the terms.

Reply:
I guess we already wrap queries to those tables, so we just need to add the resolution wrapper

Reply:
That was exactly what I was going to comment on. Should I add the wrapper to `conflict_resolution` to finalize the translation layer, or is there a different resolution we ought to discuss first?

Reply:
That sounds good, one more wrapper and at least the API-facing terminology will be consistent :+1:

Reply:
> However, there is a lot of benefit to having a [ubiquitous language](https://www.agilealliance.org/glossary/ubiquitous-language) that is shared among developers, product people, and customers - a shared vocabulary reduces friction and helps everyone be on the same page.

It's worth noting that the ubiquitous language comes *after* the bounded context. That is, it's ubiquitous *within* a context. For example, while Merlin is predominantly used in a planning context where "activities" are the unit of control, Merlin can be used in other contexts, e,g. sequence simulation, where "commands" are the unit of control. We use the term "directive" internally to encompass both of these.

Reply:
> That sounds good, one more wrapper and at least the API-facing terminology will be consistent 👍

Wrapper has been implemented. It's named `resolution_type` to avoid having to do explicit schemas whenever we convert.